sematic/ui/src/pipelines/PipelinePanels.tsx

```diff
@@ -41,16 +41,25 @@ export default function PipelinePanels() {
     return graph.runsById.get(runId) || rootRun;
   }, [selectedRunId, graph, rootRun]);
 
-  const defaultTab = selectedRun?.future_state === "FAILED" ? "logs" : "output";
-  const [selectedRunTab, setSelectedRunTab] = useState(defaultTab);
+  const [selectedTabHash, setSelectedRunTab] = useAtom(selectedTabHashAtom);
+
+  const selectedRunTab = useMemo(() => {
+    if (!!selectedTabHash) {
+      return selectedTabHash;
+    }
+    // in case there is no previously selected tab
+    const run = selectedRun || rootRun;
+    return run?.future_state === "FAILED" ? "logs" : "output";
```

Comment:
Please add a comment to this line explaining the intent of the default pane.

Reply:
Thanks. Added a comment.